config/base/src/lib.rs

```diff
@@ -83,78 +154,147 @@ pub mod derive {
     /// generating documentation, and `#[config(inner)]` for getting
     /// inner fields recursively.
     ///
+    /// # Examples
+    ///
     /// ```rust
-    /// use iroha_config_base::derive::{LoadFromEnv, Documented};
-    /// use iroha_config_base::proxy::{LoadFromEnv as _, Documented as _};
+    /// use iroha_config_base::derive::Documented;
+    /// use iroha_config_base::proxy::Documented as _;
     ///
-    /// #[derive(serde::Deserialize, serde::Serialize, LoadFromEnv, Documented)]
+    /// #[derive(serde::Deserialize, serde::Serialize, Documented)]
     /// struct Outer { #[config(inner)] inner: Inner }
     ///
-    /// #[derive(serde::Deserialize, serde::Serialize, Documented, LoadFromEnv, Debug, Clone)]
+    /// #[derive(serde::Deserialize, serde::Serialize, Documented)]
     /// struct Inner { b: String }
     ///
     /// let outer = Outer { inner: Inner { b: "a".to_owned() }};
     ///
     /// assert_eq!(outer.get_recursive(["inner", "b"]).unwrap(), "a");
     /// ```
     pub use iroha_config_derive::Documented;
+    /// Derive macro for implementing the trait
+    /// [`iroha_config::base::proxy::LoadFromDisk`](`crate::proxy::LoadFromDisk`)
+    /// trait for config structures.
+    ///
+    /// Meant to be used on proxy types only, for
+    /// details see [`iroha_config::base::derive::Proxy`](`crate::derive::Proxy`).
+    ///
+    /// The trait's only method, `from_path`,
+    /// deserializes a JSON config at the provided path into the parent proxy structure,
+    /// leaving it empty in case of any error.
+    ///
+    /// The `ReturnValue` associated type can be
+    /// swapped for anything suitable, returning just the proxy structure itself currently.
```

Comment:
```suggestion
    /// swapped for anything suitable, returning just the proxy structure itself.
```

Reply:
What I'm trying to say here is that return value could be changed, but as of now it's set to be the proxy struct.

Reply:
hm, maybe then say that by default it returns the proxy struct? something like this:

```suggestion
    /// swapped for anything suitable. Currently, the proxy structure is returned by default.
```